### PR TITLE
Stop referencing Microsoft.PowerShell.Security when the core snapin is used

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

This is a follow-up change of #16355.
The strong reference to `Microsoft.PowerShell.Security` is not needed anymore after merging #16355.

There are whitespace only changes in the PR, so it's easier to review by ignoring the whitespaces:
https://github.com/PowerShell/PowerShell/pull/17771/files?w=1

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
It seems there are more dependences on the module :-)

**Review by PaulHigin — APPROVED** [review]

**Comment by daxian-dbw** [comment]
/cc @iSazonov @SteveL-MSFT @JamesWTruher 

It turns out **it's sort-of intentional** to load `Microsoft.PowerShell.Security.dll` eagerly.
After this change (and #16355), loading of the `Microsoft.PowerShell.Security` module fails with the following error (see the [same failure in CI](https://dev.azure.com/powershell/PowerShell/_build/results?buildId=104943&view=logs&j=99798060-e5bf-55b9-b3b5-109e6acacce3&t=0050f0b1-14ee-5c4d-b4b0-da6405e3eaf0&l=1064)):

```
PS:1> Import-Module Microsoft.PowerShell.Security
Import-Module: The following error occurred while loading the extended type data file:
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(19) : Error in type "System.Security.AccessControl.ObjectSecurity": The "Type" node must have "Members", "TypeConverters", or "TypeAdapters".
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(52) : Error: CodeProperty should use a getter or setter method.
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(53) : Error: Unable to find type [Microsoft.PowerShell.Commands.SecurityDescriptorCommandsBase].
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(45) : Error: CodeProperty should use a getter or setter method.
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(46) : Error: Unable to find type [Microsoft.PowerShell.Commands.SecurityDescriptorCommandsBase].
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(38) : Error: CodeProperty should use a getter or setter method.
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(39) : Error: Unable to find type [Microsoft.PowerShell.Commands.SecurityDescriptorCommandsBase].
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(31) : Error: CodeProperty should use a getter or setter method.
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(32) : Error: Unable to find type [Microsoft.PowerShell.Commands.SecurityDescriptorCommandsBase].
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(24) : Error: CodeProperty should use a getter or setter method.
, C:\arena\Modules\Microsoft.PowerShell.Security\Security.types.ps1xml(25) : Error: Unable to find type [Microsoft.PowerShell.Commands.SecurityDescriptorCommandsBase].
```
This is because the processing of `type.ps1xml` and `format.ps1xml` happens **before** loading `NestedModules`, and hence when the `type.ps1xml` references a type from the nested module, the type cannot be resolved when processing the `type.ps1xml`.

The [use-nested-module.zip](https://github.com/PowerShell/PowerShell/files/9204179/use-nested-module.zip) is a simple repro of this behavior. Unzip it, and run `import-Module <path-to>\conflict` will demonstrate the problem.

image

However, it will work fine when using `RootModule = 'conflict.dll'` instead of `NestedModules = @('conflict.dll')` in the module manifest. I believe the root module is specially handled with eager loading. ([use-root-module.zip](https://github.com/PowerShell/PowerShell/files/9204192/use-root-module.zip))

image

Both PowerShell 7.0 and 7.2 has the same behavior. But on Windows PowerShell, both `use-nested-module` and `use-root-module` will fail to process the `type.ps1xml` file.

-------------

**QUESTION:** Changing the loading order of nested module would be a non-trivial work which is quite risky. Given that, shall we revert #16355?

**Comment by iSazonov** [comment]
@daxian-dbw Your question raise more questions:
1. Is PS team strategy to decouple built-in modules actual or it was deprecated?
2. Should we keep circular dependencies #14095?
3. Why do built-in modules use `NestedModules` instead of `RootModule` in psd1? Is it a bug?

**Comment by daxian-dbw** [comment]
A workaround is to add `RequiredAssemblies = "Microsoft.PowerShell.Security.dll"` in addition to the `NestedModules`. Required assemblies are always loaded before processing type/format files. I have pushed a commit with that change. Although it looks weirdly redundant, I think it should be acceptable.

File: src/Modules/Windows/Microsoft.PowerShell.Security/Microsoft.PowerShell.Security.psd1 [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+CmdletsToExport = "Get-Acl", "Set-Acl", "Get-PfxCertificate", "Get-Credential", "Get-ExecutionPolicy", "Set-ExecutionPolicy", "Get-AuthenticodeSignature", "Set-AuthenticodeSignature", "ConvertFrom-SecureString", "ConvertTo-SecureString", "Get-CmsMessage", "Unprotect-CmsMessage", "Protect-CmsMessage" , "New-FileCatalog" , "Test-FileCatalog"
 AliasesToExport = @()
-NestedModules="Microsoft.PowerShell.Security.dll"
-TypesToProcess="Security.types.ps1xml"
+NestedModules = "Microsoft.PowerShell.Security.dll"
+# 'Security.types.ps1xml' refers to types from 'Microsoft.PowerShell.Security.dll' and thus requiring to load the assembly before processing the type file.
+# We declare 'Microsoft.PowerShell.Security.dll' in 'RequiredAssemblies' so as to make sure it's loaded before the type file processing.
+RequiredAssemblies = "Microsoft.PowerShell.Security.dll"
```
Comment (iSazonov) [inline src/Modules/Windows/Microsoft.PowerShell.Security/Microsoft.PowerShell.Security.psd1:15]: Great!

**Review by iSazonov — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]
The workaround with `RequiredAssemblies` is low risk and I think this is a change we should have to decouple them

**Pull request merged**